### PR TITLE
#578: Update to FFmpeg 7.1

### LINKED ISSUE #578: [Maintenance]: Update to FFmpeg 7.0.2
Update OpenRV to use the latest version of FFmpeg

### PR DESCRIPTION
**Pull request by williamwira**
### Linked issues
Fixes #578

### Summarize your change.
Upgraded FFmpeg to 7.1, and resolved build-time errors and deprecation warnings.
Added build flag RV_FFMPEG so app builder can decide between FFmpeg 6 and FFmpeg 7. (FFmpeg 6 is default)

### Describe the reason for the change.
Maintenance Update

### Describe what you have tested and on which operating system.
MacOS Sonoma 14.7
Linux RHEL 9.2

### Add a list of changes, and note any that might need special attention during the review.
MovieFFMpeg
ffmpeg.cmake

Build with no RV_FFMPEG flag to build against FFmpeg 6, or do `rvcfg -DRV_FFMPEG=7` to use FFmpeg 7.
After building: Open RV, load a QuickTime video file. Confirm the video loads and can be played.

### CODE REVIEW

**Review by bernie-laberge — APPROVED** [review]
LGTM
Thank you for this excellent contribution to Open RV Will !

File: cmake/defaults/cxx_defaults.cmake [hunk 1]
```diff
@@ -72,3 +72,19 @@ ELSEIF(RV_VFX_PLATFORM STREQUAL CY2023)
   SET(RV_VFX_CY2023 ON)
   ADD_COMPILE_DEFINITIONS(RV_VFX_CY2023)
 ENDIF()
+
+#
+# FFMpeg option
```
Comment (bernie-laberge) [inline cmake/defaults/cxx_defaults.cmake:77]: Thank you for this excellent contribution to the cause Will ! 
My only comment is cosmetic: perhaps we should try to be consistent and use FFmpeg for the name.
In accordance with the official naming: https://en.wikipedia.org/wiki/FFmpeg
So here, replace FFMpeg with FFmpeg

File: cmake/defaults/cxx_defaults.cmake [hunk 2]
```diff
@@ -72,3 +72,19 @@ ELSEIF(RV_VFX_PLATFORM STREQUAL CY2023)
...
+#
+
+# Add preprocessor variable for use in code and
+# cmake to determine the current version of ffmpeg.
```
Comment (bernie-laberge) [inline cmake/defaults/cxx_defaults.cmake:81]: cosmetic chance: 
replace ffmpeg with FFmpeg

File: cmake/defaults/rv_options.cmake [hunk 3]
```diff
@@ -76,4 +76,38 @@ SET(RV_VFX_PLATFORM
 SET_PROPERTY(
   CACHE RV_VFX_PLATFORM
   PROPERTY STRINGS ${_RV_VFX_PLATFORM}
-)
\ No newline at end of file
+)
+
+#
+# FFMpeg option
```
Comment (bernie-laberge) [inline cmake/defaults/rv_options.cmake:82]: cosmetic chance: 
replace FFMpeg with FFmpeg

File: cmake/defaults/rv_options.cmake [hunk 4]
```diff
@@ -76,4 +76,38 @@ SET(RV_VFX_PLATFORM
...
+# This option is to control the version of ffmpeg
+# that OpenRV downloads and installs.
+#
+# There will be one version used by default per
+# major release. Any other version will require
+# RV_FFMPEG env var to be set with the desired
+# major version.
+#
+
+# Supported FFMpeg versions.
```
Comment (bernie-laberge) [inline cmake/defaults/rv_options.cmake:93]: cosmetic chance: 
replace FFMpeg with FFmpeg

File: cmake/dependencies/ffmpeg.cmake [hunk 5]
```diff
@@ -23,17 +23,33 @@ SET(_target
...
+  SET(_version
+      "n6.1.1"
+  )
+
+  SET(_download_hash
+      "6dfc27fcb6da6f653c6ec025c2cd9b00"
+  )
+ELSE()
+  # This shouldn't happen, but it could if there are changes the acceptable
+  # versions of ffmpeg in ffmpeg.cmake and this section is not
```
Comment (bernie-laberge) [inline cmake/dependencies/ffmpeg.cmake:44]: cosmetic chance:
replace ffmpeg with FFmpeg

**Review by cedrik-fuoco-adsk — APPROVED** [review]

**Review by eloisebrosseau — APPROVED** [review]
Small nitpick, but this is really great! Thank you Will!

File: src/lib/image/MovieFFMpeg/MovieFFMpeg.cpp [hunk 6]
```diff
@@ -2699,13 +2701,15 @@ MovieFFMpegReader::idAudioChannels(uint64_t layout, int numChannels)
...
         }
 
-        const char *chName, *chDesc;
-        chName = av_get_channel_name(unmasked);
-        chDesc = av_get_channel_description(unmasked);
-
-        DBL (DB_AUDIO, "Audio ch " << (chans.size()-1)
-                    << " is: '" << chName
-                    << "-" << chDesc);
+        AVChannel ch = av_channel_layout_channel_from_index(&layout, i);
```
[the remark this answers is not in the document]
Comment (williamwira) [reply src/lib/image/MovieFFMpeg/MovieFFMpeg.cpp:2704]: Definitely, thanks @eloisebrosseau ! Also noticed I wasn't handling return values of `av_channel_name` and `av_channel_description` correctly, they are valid as long as the returned int is > 0. I'll fix that as well.

[the remark this answers is not in the document]
Comment (williamwira) [reply src/lib/image/MovieFFMpeg/MovieFFMpeg.cpp]: I pushed an update, let me know what you think. Thanks!

[the remark this answers is not in the document]
Comment (eloisebrosseau) [reply src/lib/image/MovieFFMpeg/MovieFFMpeg.cpp]: It's perfect, thank you!

**Pull request merged**